docs/pages/dev/reference/API/org_unit_registry.md

```diff
@@ -0,0 +1,436 @@
+
+# POST /api/orgunits/changes/?app_id=..
+
+API used to create or modify an OrgUnit with validation (see feature flags)
+When creating an OrgUnit, the client doesn't know the ID or the OrgUnit was not created yet and we might want to apply 
+some changes still. Therefore, the IDs can be either the DB id or the UUID generated by the client.
+Changes that are not provided are considered not being part of the change request.
+
+## Permissions
+
+User must be authenticated.
+
+## Query parameters
+
+- app_id: String - Must be provided, project for which this is created.
+
+## Body
+
+```json
+{
+  "uuid": "UUID - Client generated UUID",
+  "org_unit_id": "String - id or UUID of the OrgUnit to change, if UUID doesn't exist in DB, it's a new OrgUnit.",
+  "created_at": "Timestamp - format same as /api/instances",
+  "updated_at": "Timestamp - format same as /api/instances",
+  "changes": { "": "Changes is not null and must contain at least one field",
+    "parent_id": "String? - id or UUID of the parent OrgUnit, may be null or omitted.",
+    "name": "String? - Name of the OrgUnit, may be null or omitted.",
+    "status": "Enum<Status>? - One of `open` or `close`",
+    "org_unit_type_id": "Int? - id of the OrgUnitType, may be null or omitted",
+    "groups": "Array of Group ids? - can be empty, null or omitted. Empty means we want to remove all values",
+    "geo_point": { "": "New geopoint for the OrgUnit, may be null or omitted",
+      "latitude": "Double - New latitude of the OrgUnit",
+      "longitude": "Double - New longitude of the OrgUnit",
+      "altitude": "Double - New altitude of the OrgUnit",
+      "accuracy": "Double - New accuracy of the OrgUnit"
+    },
+    "geo-shape": "String? - GeoJSON representation, may be null or omitted",
+    "forms": [ "Array of form objects? - may be null or omitted, cannot be empty",
+      {
+        "form_id": "id of the form",
+        "instance_id": "UUID of the instance"
+      }
+    ]
+  }
+}
+```
+
+## Possible responses
+
+### 201 - Created
+
+### 400 - Bad request
+
+- `app_id` was not provided
+- A not nullable field was null or omitted
+- `org_unit_id` is a new OrgUnit and the following fields were null or omitted: `parent_id`, `name`, `geo_point` and `org_unit_type_id`
+- A `String` field has an empty value
+- `parent_id` is not a valid OrgUnit
+- `status` was not one of the enum values.
+- `org_unit_type_id` is not a valid OrgUnitType
+- One of the `groups` id is not a valid Group
+- `forms` is empty
+- `form_id` is not a reference form for the OrgUnitType
+- `org_unit_type_id` is not a valid sub OrgUnitType for the given parent --> TO BE CONFIRMED
+
+### 401 - Unauthorized
+
+- No authentication token or an invalid one was provided
+
+### 403 - Forbidden
+
+- `org_unit_type_id` is not a valid OrgUnitType to be created for the given `parent_id`
+- `org_unit_type_id` is not a valid OrgUnitType to be moved to the given `parent_id`
+
+### 404 - Not found
+
+- `form_id` is not found
```

Comment:
Ideally, the API should indicate which field is creating the 404 (this is something that we're lacking in other endpoints)